test/integration2/dialects/mysql2.spec.js

```diff
@@ -84,5 +84,30 @@ describe('MySQL dialect', () => {
         });
       });
     });
+
+    dbs.forEach((_db) => {
+      describe('Regressions', () => {
+        it('should not modify passed config', (done) => {
+          const mysql = require('mysql');
+
+          // Config copied from test/integration2/util/knex-instance-provider.js
+          const config = {
+            client: 'mysql2',
+            connection: getTestConnectionConfig('mysql2'),
+          };
+
+          // This is the step that should NOT modify the `config` object
+          const knex = require('../../../lib/index')(config);
+          knex.destroy();
+
+          const connection = mysql.createConnection(config.connection);
+          connection.connect(function (err) {
+            expect(err).to.be.null;
+            connection.end();
+            done()
+          })
+        })
+      })
```

Comment:
This test does not fail in master, please write a failing test so that we can avoid future bugs

Reply:
hey @castarco, i have a test associated to my own [branch](https://github.com/manufacturedba/knex/tree/regression-check) that could be substituted here. it fails successfully without your library changes

relevant commit - https://github.com/manufacturedba/knex/commit/7738705fe506aa75307dd0e8ba15f2100554240c

can also kick off another PR if you don't have time